apps/forklift/lib/forklift/jobs/job_utils.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Forklift.Jobs.JobUtils do
+  @moduledoc false
+  use Retry.Annotation
+  alias Pipeline.Writer.TableWriter.Helper.PrestigeHelper
+
+  @retries Application.get_env(:forklift, :compaction_retries, 10)
+  @backoff Application.get_env(:forklift, :compaction_backoff, 10)
+
+  @retry with: constant_backoff(@backoff) |> Stream.take(@retries)
+  def verify_count(table, target_count, message) do
+    with {:ok, actual_count} <- PrestigeHelper.count(table),
+         {:ok, _} <- check_count(actual_count, target_count) do
```

Comment:
could this just be true <- actual_count == target_count?

I'm unsure if this is better but its less code?